### ISSUE TITLE
With TLP: Cannot access secondary GPU - error: Could not enable discrete graphics card

### ISSUE THREAD

**Issue by jkdf2** [Bumblebee-Project/Bumblebee#829]
TLP disables the GPU driver and you get `[ERROR]Cannot access secondary GPU - error: Could not enable discrete graphics card
`. This has been fixed by a user on the Arch forums: fix is to blacklist the GPU driver with `RUNTIME_PM_BLACKLIST` and `RUNTIME_PM_DRIVER_BLACKLIST`:

https://bbs.archlinux.org/viewtopic.php?id=218757

Filing a bug report with TLP so that it can perhaps detect an Nvidia Optimus configuration.

**Comment by jkdf2** [comment]
Help needed for upstream patch on TLP, see open GitHub issue linked above.

**Comment by linrunner** [comment]
> TLP disables the GPU driver and you get 
Wrong. TLP doesn't touch drivers. TLP enables Runtime PM for all PCIe devices – including the eGPU.

**Comment by jkdf2** [comment]
> TLP enables Runtime PM for all PCIe devices – including the eGPU.

Oops, thanks for the clarification. Does this mean there might be a better approach than to blacklist the card & driver from Runtime PM?

**Comment by ArchangeGabriel** [comment]
The better approch is probably fixing bbswitch to be a proper driver. Once that is done, not sure whether blacklisting it would be necessary, but I suppose not: after all, bbswitch would be a driver only managing runtime PM of the nvidia card.

**Comment by Lekensteyn** [comment]
I have noticed that when the dGPU is powered off (via PCIe port PM in Linux 4.8+) while no driver is loaded, nouveau fails to load. It might also cause this issue with the blob, but I have not checked.

**Comment by eXt73** [comment]
;) Try my APM 5.2 [Advanced Power&Performance managment] ... working great with my build od bumble ;) > www.netext73.pl

**Comment by ashic** [comment]
I'm running Ubuntu 16.10 with kernel 4.10.2. Bumblebee is at 3.2.1, nvidia 378 is installed. optirunning any command gives me this error. I removed the 32 bit libs from bumblebee config, but the issue doesn't go away, even after fresh reboot. I also see that I don't have a file at /etc/default/tlp. Any pointers?

**Comment by ashic** [comment]
It seems using primusrun, things work perfectly, but with optirun, I get that error.

**Comment by linrunner** [comment]
> I don't have a file at /etc/default/tlp.
@ashic: you don't use TLP, so your postings are offtopic here.
